portfolio/src/main/webapp/tasksApi.js

```diff
@@ -0,0 +1,169 @@
+var GoogleAuth;
+
+// Scope for read access to Tasks API
+const SCOPE_READ_TASKS = 'https://www.googleapis.com/auth/tasks.readonly';
+
+const CLIENT_ID_TASKS =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+const DISCOVERY_DOCS_TASKS =
+    'https://www.googleapis.com/discovery/v1/apis/tasks/v1/rest';
+
+var API_KEY_TASKS = '';
+
+function fetchApiKey() {
+  fetch('./appConfigServlet')
+      .then(response => response.json())
+      .then((responseJson) => {
+        API_KEY_TASKS = responseJson['API_KEY'];
+      });
+}
+
+function handleClientLoadTasks() {
+  // Load the API's client and auth2 modules.
+  // Call the initClient function after the modules load.
+  gapi.load('client:auth2', initClientTasks);
+}
+
+function initClientTasks() {
+  fetchApiKey();
+
+  // Initialize the gapi.client object, which app uses to make API requests.
+  // Get API key and client ID from API Console.
+  // 'scope' field specifies space-delimited list of access scopes.
+  gapi.client
+      .init({
+        'apiKey': API_KEY_TASKS,
+        'clientId': CLIENT_ID_TASKS,
+        'discoveryDocs': [DISCOVERY_DOCS_TASKS],
+        'scope': SCOPE_READ_TASKS
+      })
+      .then(function() {
+        GoogleAuth = gapi.auth2.getAuthInstance();
+
+        // Listen for sign-in state changes.
+        GoogleAuth.isSignedIn.listen(updateSigninStatus);
+
+        // Handle initial sign-in state. (Determine if user is already signed
+        // in.)
+        updateSigninStatus(GoogleAuth.isSignedIn.get());
+      });
+}
+
+function handleAuthClick() {
+  if (GoogleAuth.isSignedIn.get()) {
+    // User is authorized and has clicked "Sign out" button.
+    GoogleAuth.signOut();
+  } else {
+    // User is not signed in. Start Google auth flow.
+    GoogleAuth.signIn();
+  }
+}
+
+function updateSigninStatus(isSignedIn) {
+  if (isSignedIn) {
+    drawImportMenu();
+  } else {
+    clearImportMenu();
+  }
+  updateButtonText(isSignedIn);
+}
+/**
+ *Iterate through all the user's tasklists and pass
+ *them to updateTaskList().
+ */
+function importAllTasks() {
+  if (GoogleAuth.isSignedIn.get()) {
+    gapi.client.tasks.tasklists.list({'maxResults': 100})
+        .then(function(response) {
+          var taskLists = response.result.items;
+          taskLists.forEach(tasklist => {
```

Comment:
Just noticed while testing locally: if the taskLists is undefined, this line might throw an error.

Reply:
Done.